src/System.Private.CoreLib/shared/System/Diagnostics/Contracts/ContractException.cs

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+
+namespace System.Diagnostics.Contracts
+{
+    [Serializable]
+    [System.Runtime.CompilerServices.TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
+    // Needs to be public to support binary serialization compatibility
+    public sealed class ContractException : Exception
+    {
+        private readonly ContractFailureKind _kind;
+        private readonly string _userMessage;
+        private readonly string _condition;
+
+        public ContractFailureKind Kind { get { return _kind; } }
+        public string Failure { get { return this.Message; } }
+        public string UserMessage { get { return _userMessage; } }
+        public string Condition { get { return _condition; } }
+
+        // Called by COM Interop, if we see COR_E_CODECONTRACTFAILED as an HRESULT.
+        private ContractException()
+        {
+            HResult = System.Runtime.CompilerServices.ContractHelper.COR_E_CODECONTRACTFAILED;
+        }
+
+        public ContractException(ContractFailureKind kind, string failure, string userMessage, string condition, Exception innerException)
+            : base(failure, innerException)
+        {
+            HResult = System.Runtime.CompilerServices.ContractHelper.COR_E_CODECONTRACTFAILED;
+            _kind = kind;
+            _userMessage = userMessage;
+            _condition = condition;
+        }
+
+        private ContractException(System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context)
```

Comment:
Nit: The common style for exceptions is to have `using System.Runtime.Serialization;`